Add retry logic for network failures

nsmith22 opened the pull request:
# Description of change
Allows for 3 network retries with a simple linear back-off algorithm. The code performs a retry upon any exception that comes out of the `requests` package.

# Manual QA steps
Updated the v2 `executor` to point at the commit hash for this change and ran the github tap for one repo successfully. Then, ran the repo again but without internet connectivity in order to validate the retry logic.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

KBorders01 reviewed (approved):
Looks good, though perhaps we should increase the limit above three so it will retry for a few minutes?

nsmith22 commented:
> Looks good, though perhaps we should increase the limit above three so it will retry for a few minutes?

Updated retry count to 5 and changed the linear backoff to be (retry * 30s) in order to get several minutes of retries.
